internal/pkg/runtime/engines/singularity/process_linux.go

```diff
@@ -145,8 +145,69 @@ func (engine *EngineOperations) checkExec() error {
 	return fmt.Errorf("no %s found inside container", args[0])
 }
 
+// startFuseDrivers starts the required FUSE drivers
+func startFuseDrivers(engine *EngineOperations) error {
+	for _, name := range engine.EngineConfig.GetPluginFuseMounts() {
+		var cfg struct {
+			Fuse struct {
+				DevFuseFd  int
+				MountPoint string
+				Program    []string
+			}
+		}
+
+		if err := engine.EngineConfig.GetPluginConfig(name, &cfg); err != nil {
+			return err
+		}
+
+		fh := os.NewFile(uintptr(cfg.Fuse.DevFuseFd), "fd-"+name)
+		fd := fh.Fd()
+		// The assumption is that the plugin prepared "Program"
+		// in such a way that it's missing the last parameter
+		// and that must correspond to /dev/fd/N
+		args := append(cfg.Fuse.Program, fmt.Sprintf("/dev/fd/%d", fd))
+		cmd := exec.Command(args[0], args[1:]...)
+
+		// Add the /dev/fuse file descriptor to the list of file
+		// descriptors to be passed to the new process.
+		//
+		// ExtraFiles is an array of *os.File, with the position
+		// of each entry determining the resulting file
+		// descriptor number. It must correspond to the file
+		// descriptor we have so that the above /dev/fd/%d
+		// works. Use fd-3 because we must skip over stdin,
+		// stdout, stderr; +1 because the second argument to
+		// make is the number of elements.
+		cmd.ExtraFiles = make([]*os.File, fd-3+1)
```

Comment:
What if you just pass `fh` to `ExtraFiles` and set `/dev/fd/3` above instead ?

Reply:
Yeah, that should work, thanks.